GVRf/Framework/src/org/gearvrf/GVRCameraRig.java

```diff
@@ -156,6 +159,8 @@ public float getFloat(String key) {
      *            The {@code float} value to map.
      */
     public void setFloat(String key, float value) {
+        checkKeyNotNullOrEmpty(key);
+        checkNotNaN(value);
```

Comment:
We might want to use infinities to hold error codes - `{NaN, +INF, -INF}` is three times as many error codes as `{NaN}`.

>  I am trying to balance the overhead of this class and how many conditions we should check

Fair enough. But once we're calling static methods in a different package, I don't know that `Float.isNan(value) || Float.isInfininite(value)` is significantly more expensive than `Float.isNan(value)`.